### ISSUE TITLE
Memory leak when click on notification.

### ISSUE THREAD

**Issue by iceteahh** [googlecast/CastCompanionLibrary-android#198]
This happen to my project and your sample application.

My scenario is:
1. Connect to `ChromeCast` device
2. Cast a video to `ChromeCast` -> `VideoCastControllerActivity` show up.
3. Press home button -> Notification show up.
4. Press the notification to return `VideoCastControllerActivity`
5. Memory are growing...

Loop 3 + 4, and RAM increase continuity .

My manifest file:
```
<activity                android:name="com.google.android.libraries.cast.companionlibrary.cast.player.VideoCastControllerActivity"
            android:label="@string/app_name"
            android:launchMode="singleTask"
            android:parentActivityName=".MainActivity"
            android:screenOrientation="portrait">
            <meta-data
                android:name="android.support.PARENT_ACTIVITY"
                android:value=".MainActivity" />
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />
            </intent-filter>
</activity>
<activity
            android:name="com.flix.apps.android.MainActivity"
            android:configChanges="keyboard|keyboardHidden|screenSize|orientation"
            android:label="@string/app_name"
            android:launchMode="singleTask" />
```

**Comment by naddaf** [comment]
Is this reproducible in the latest version (version 2.3) with CastVideos-android app?

**Comment by iceteahh** [comment]
Yes i am using your latest library and sample app version.

**Comment by naddaf** [comment]
No, you were not; that library was updated a few minutes ago after you had opened your issue, so you couldn't have possibly used v2.3

**Comment by iceteahh** [comment]
You right, my mistake. But does it fix in new version?

**Comment by naddaf** [comment]
I don't know, I am asking you to test with the latest one and CastVideos-android app. If you can reproduce, the let me know.

**Comment by iceteahh** [comment]
Ok, i will test your latest version. Thank for your response.

**Comment by naddaf** [comment]
It may very well be present there too but the last update had made a fair
amount of changes in the notifications so it is good to have it tested
again, thanks.

**Comment by iceteahh** [comment]
I have tested your latest sample app and library and still have this problem. Here is your sample app's memory graph.
![image](https://cloud.githubusercontent.com/assets/5756363/9323572/29b169ec-45ac-11e5-9adc-15c94325913d.png)

**Comment by naddaf** [comment]
Thanks for the report. I have fixed this in 2.3.1.

**Comment by Nimrodda** [comment]
Hi @naddaf I actually made a pull request that fixed this issue **before** you made your commit, as seen in this issue's time of events. If you don't accept pull requests, you should either block the feature from this repository or mention it somewhere the the README file.

This issue is critical, are you planning to make a hotfix release anytime soon?

**Comment by naddaf** [comment]
Nimrodda,
Not sure by your last sentence about hotfix, this issue is fixed in the code base. As for accepting pull requests, I usually look at the issue itself to see what it is and sometimes I use the pull requests and sometimes I don't; I often have to fix things myself since there is a large number of things that I might be aware of, related to various projects that are using the library or similar things, that I often end up fixing things myself since the suggested pull request doesn't cover all the corner cases that are needed or can cause issues elsewhere, etc. What I always do is that after coming up with my fix, I look at the pull request to make sure I haven't missed anything and then do some tests and make the fix. So in general, I do appreciate people sending me pull requests but I hope there is no expectation to always accept them.

Thanks again.

**Comment by Nimrodda** [comment]
Glad I could help! :) Nevertheless, it would be nice to get your feedback on the pull request and if needed improvements can always be made. Thanks and great library overall! :+1:

**Comment by naddaf** [comment]
Nimrod,

Your pull request was good, I can't see any issue in there, so great work
and good catch. When I saw your email that the leak exists in the recent
version as well, I was wondering if it had to do anything with the
notifications at all since there was nothing special there that could have
caused that so I removed that from the reproduction steps and just started
opening and closing the VideoCastControllerActivity and noticed the memory
leak. In these cases, memory leaks usually come from either holding on to
resources (such as images) or are commonly caused by not releasing a
listener, etc ad a quick look at the code showed me I was not releasing the
mini controller; I changed that and tested and it was fixed. Then I noticed
your pull request and your data and fix backed up what I had found so I was
more confident in the fix.

Again, thanks for reporting and further looking into a solution, I always
benefit and learn from folks that spend time and contribute to this project.

As for using onPause vs onStop or onDestroy, there is very little (if any)
difference in this case; the important thing is to make sure it is
registered every time and unregistered when we are done; I often use
onPause, if possible, rather than onDestroy so that when my activity is not
visible (and still not destroyed), I don't use unneeded resources but
again, I don't think there is much a difference specially for that
activity. Are you using this in your own app, and if so, what app is that?
